tests/test_htcondor_service.py

```diff
@@ -1147,5 +1218,62 @@ def testSuccess(self):
         self.assertIn("check_group_status.sub", job.subfile)
 
 
+class CreatePeriodicReleaseExprTestCase(unittest.TestCase):
+    """Test _create_periodic_release_expr function."""
+
+    def testNoReleaseExpr(self):
+        results = htcondor_service._create_periodic_release_expr(2048, 1, 32768, "")
+        self.assertEqual(results, "")
+
+    def testJustMemoryReleaseExpr(self):
+        self.maxDiff = None  # so test error shows entire strings
+        results = htcondor_service._create_periodic_release_expr(2048, 2, 32768, "")
+        truth = (
+            "JobStatus == 5 && NumJobStarts <= JobMaxRetries && "
+            "(HoldReasonCode =?= 34 && HoldReasonSubCode =?= 0 || "
+            "HoldReasonCode =?= 3 && HoldReasonSubCode =?= 34) && "
+            "min({int(2048 * pow(2, NumJobStarts - 1)), 32768}) < 32768"
+        )
+        self.assertEqual(results, truth)
+
+    def testJustUserReleaseExpr(self):
+        results = htcondor_service._create_periodic_release_expr(2048, 1, 32768, "True")
+        truth = "JobStatus == 5 && NumJobStarts <= JobMaxRetries && HoldReasonCode =!= 1 && True"
+        self.assertEqual(results, truth)
+
+    def testMemoryAndUserReleaseExpr(self):
+        self.maxDiff = None  # so test error shows entire strings
+        results = htcondor_service._create_periodic_release_expr(2048, 2, 32768, "True")
+        truth = (
+            "JobStatus == 5 && NumJobStarts <= JobMaxRetries && "
+            "((HoldReasonCode =?= 34 && HoldReasonSubCode =?= 0 || "
+            "HoldReasonCode =?= 3 && HoldReasonSubCode =?= 34) && "
+            "min({int(2048 * pow(2, NumJobStarts - 1)), 32768}) < 32768 || "
+            "HoldReasonCode =!= 1 && True)"
+        )
+        self.assertEqual(results, truth)
```

Comment:
As the `multiplier` might be **None** it would be good to have at least one test that covers this case.